docs/src/main/sphinx/develop/table-functions.rst

```diff
@@ -58,11 +58,49 @@ identified by the catalog name, the schema name, and the function name.
 The function can use the schema name, for example to use data from the
 indicated schema, or ignore it.
 
-- function name
-- list of expected arguments
+- **function name**
+- **list of expected arguments**
 
-You can specify default values for some arguments, and those arguments can be
-skipped during invocation:
+Three different types of arguments are supported: scalar arguments, descriptor
+arguments, and table arguments. See :ref:`argument_types` for details. You can
+specify default values for scalar and descriptor arguments. The arguments with
+specified default can be skipped during table function invocation.
+
+- **returned row type**
+
+It describes the row type produced by the table function.
+
+If a table function takes table arguments, it can additionally pass the columns
+of the input tables to output using the ``pass-through mechanism``. The
+returned row type is supposed to describe only the columns produced by the
+function, as opposed to the pass-through columns.
+
+In the example, the returned row type is ``GENERIC_TABLE``, which means that
+the row type is not known statically, and it is determined dynamically based on
+the passed arguments.
+
+When the returned row type is known statically, you can declare it using:
+
+.. code-block:: java
+
+    new DescribedTable(descriptor)
+
+If a table function does not produce any columns, and it only outputs the
+pass-through columns, use ``ONLY_PASS_THROUGH`` as the returned row type.
+
+Note: a table function must return at least one column. It can either be a
+proper column, i.e. produced by the function, or a pass-through column.
+
+.. _argument_types:
```

Comment:
all ref anchors share a global namespace in the docs .. should we add some prefix for 
this page like _tf_argument_type